Fix: Preserve GET parameters in API calls

DanieleAlessandra opened the pull request:
Added logic to append the query string to the $pCanonizedPath in cases where the API method is GET and $pParams is not empty. This ensures that the GET parameters are correctly passed and not lost during the API call.

DanieleAlessandra committed:
Manage GET params from input Array

DanieleAlessandra committed:
Added Unit Test

swashata commented on .gitattributes:
```diff
@@ -0,0 +1,7 @@
+# Items to ignore when downloading a zip
+.git export-ignore
```
`.git` is already ignored. Please correct me if I am wrong though.

swashata commented on .gitignore:
```diff
@@ -0,0 +1,3 @@
+.phpunit.result.cache
```
Also missing the `vendor` directory that will be created by composer.

swashata commented on freemius/Freemius.php:
```diff
@@ -251,6 +251,9 @@ public function MakeRequest($pCanonizedPath, $pMethod = 'GET', $pParams = array(
                     }
 
                     $opts[CURLOPT_RETURNTRANSFER] = true;
+                } else if ( 'GET' === $pMethod && ! empty( $pParams ) ) {
```
Please take a look at the rest of the file :) We are not using WP coding standard, rather Freemius. So please adjust accordingly

```php
}
else if ('GET' === $pMethod && ! emoty($pParams)
{
    $pCanonizedPath = $this->AddQueryParams($pCanonizedPath, $pParams);
}
```

Also good to extract the `AddQueryParams` to the base abstract class (since we already have that).

swashata commented on phpunit.xml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit bootstrap="vendor/autoload.php"
```
Please create a file under

* `tests/bootstrap.php`

This will load the `vendor/autoload.php` and also the `credentials.php` file along with the library itself.

This way we wouldn't have to include the library again when writing new tests.

swashata commented on tests/GetLicenseDataTest.php:
```diff
@@ -0,0 +1,42 @@
+{
+    public function testLicenseData()
+    {
+        require_once 'credentials.php';
```
Please see my suggestion above, we can do it from a `bootstrap.php` file under `tests` directory.

swashata commented on tests/GetLicenseDataTest.php:
```diff
@@ -0,0 +1,42 @@
+ * How to set up and run tests:
+ *
+ * 1. Install PHPUnit via Composer:
+ *    composer require --dev phpunit/phpunit
```
You already have installed with `composer.json`. So one just needs to do `composer install`.

swashata commented on tests/GetLicenseDataTest.php:
```diff
@@ -0,0 +1,42 @@
+            FS__API_SECRET_KEY
+        );
+
+        $license_key = 'sk_rr_s^T6x3aT^vpE-teFhtgZXjjhk3';
```
The license key also needs to come through a constant or environment variable. For simplicity, let's have it in a constant like `FS__TEST_VALID_LICENSE_KEY`. Otherwise one would need to edit the test before running it.

swashata commented on tests/GetLicenseDataTest.php:
```diff
@@ -0,0 +1,42 @@
+            'search' => ($license_key),
+        ));
+
+        $this->assertIsArray($result->licenses, 'Licenses should be an array');
```
Nice 👏

swashata commented on tests/credentials.example.php:
```diff
@@ -0,0 +1,8 @@
+
+define('FS_API__ADDRESS', 'http://api.freemius-local.com:8080');
+define('FS__API_SCOPE', 'plugin'); // e.g., 'plugin', 'theme', 'developer'
+define('FS__API_ID', 115); // Your API ID
```
Please rephrase comment to

```
The ID of the scope entity
```

Similarly "The public key of the scope entity" and "The secret key of the scope entity".

DanieleAlessandra committed:
Added AddQueryParams method to FreemiusBase
Added boostrap.php file
Removed License Key from test code
Code conventions and formatting

DanieleAlessandra committed:
Added AddQueryParams method to FreemiusBase
Added boostrap.php file
Removed License Key from test code
Code conventions and formatting

DanieleAlessandra committed:
Merge branch 'feature/filter-license-data' of github.com:Freemius/freemius-php-sdk into feature/filter-license-data

* 'feature/filter-license-data' of github.com:Freemius/freemius-php-sdk:
  Added AddQueryParams method to FreemiusBase Added boostrap.php file Removed License Key from test code Code conventions and formatting

# Conflicts:
#	tests/credentials.example.php

swashata committed:
[documentation] [license] [minor] Minor comment update + Moved license into the root.

swashata reviewed (approved):
Nicely done 👏
